src/utils.ts

```diff
@@ -149,6 +154,135 @@ export function extensionConfigurationSet<
+    : "c10999081cef8b56ea164d9a170470c268045254829e48bf4ac137a07a8a874c"
+}
+
+export function findLanguageServer(): string | null {
+  const platform = os.platform();
+  if (platform != "win32" && platform != "darwin") {
+    return null;
+  }
+  const arch = os.arch();
+  if (arch != "x64") {
+    return null;
+  }
+  const lspDir = path.join(getExtensionDir(), "lsp");
+  const filename = platform == "win32" ? "Swift-MesonLSP.exe" : "Swift-MesonLSP";
+  const fullpath = path.join(lspDir, filename);
+  if (fs.existsSync(fullpath)) {
+    return fullpath;
+  }
+  return null;
+}
+
+export async function setupLanguageServer(client: MesonLanguageClient) {
+  const lspDir = path.join(getExtensionDir(), "lsp");
+  fs.rmSync(
+    lspDir,
+    {
+      recursive: true,
+      force: true
+    }
+  )
+  fs.mkdirSync(
+    lspDir,
+    { recursive: true }
+  )
+  const tmpPath = path.join(os.tmpdir(), `lsp-${Date.now()}.zip`);
+  try {
+    const promise = downloadFile(createDownloadURL(), tmpPath);
+    promise.then(() => {
+      const hashPromise = computeFileHash(tmpPath);
+      hashPromise.then(str => {
+        const expected = createHashForLanguageServer();
+        if (str != expected) {
+          vscode.window.showErrorMessage(`Bad hash: Expected ${expected}, got ${str}!`);
+          return;
+        }
+        const zip = new Admzip(tmpPath);
+        zip.extractAllTo(lspDir);
+        if (os.platform() != "win32") {
+          fs.chmodSync(path.join(lspDir, "Swift-MesonLSP"), 0o755);
+        }
+        vscode.window.showInformationMessage("Language server is setup correctly!");
+        fs.unlinkSync(tmpPath);
+        client.restart();
+      }).catch((err: Error) => vscode.window.showErrorMessage(err.message));
+    }).catch((err: Error) => vscode.window.showErrorMessage(err.message));
+
+  } catch (err) {
+    vscode.window.showErrorMessage(JSON.stringify(err));
+    return;
+  }
+}
+
+
+async function downloadFile(url: string, dest: string): Promise<void> {
+  return new Promise((resolve, reject) => {
+    const file = fs.createWriteStream(dest);
+    const request = https.get(url, (response) => {
+      if (response.statusCode === 302 || response.statusCode === 301) {
+        const redirectUrl = response.headers.location;
+        downloadFile(redirectUrl, dest)
+          .then(() => resolve())
+          .catch((err) => reject(err));
+      } else {
+        response.pipe(file);
+        file.on('finish', () => {
+          file.close();
+          resolve();
+        });
+        file.on('error', (err) => {
+          vscode.window.showErrorMessage(`Error writing to file: ${err}`);
+          reject(err);
+        });
+      }
+    });
+    request.on('error', (err) => {
+      vscode.window.showErrorMessage(`Request error: ${err}`);
+      reject(err);
+    });
+  });
+}
+
+const getExtensionDir = (): string => {
+  return path.join(os.homedir(), ".vscode-meson");
```

Comment:
Doesn't VSCode give extensions a place to store data?

Reply:
I'm sure there is one, but I didn't find it :)

Reply:
https://code.visualstudio.com/api/extension-capabilities/common-capabilities#data-storage

Reply:
Yes, I found it already ^^ Nevertheless thanks